usecases/modules/vectorizer.go

```diff
@@ -206,6 +218,9 @@ func (p *Provider) batchUpdateVector(ctx context.Context, objects []*models.Obje
 		}
 		vectors, addProps, vecErrors := vectorizer.VectorizeBatch(ctx, objects, skipRevectorization, cfg)
 		for i := range objects {
+			if ctx.Err() != nil {
+				return nil, fmt.Errorf("context is done: %w", ctx.Err())
+			}
```

Comment:
same here: there is nothing expensive happening in this loop, I don't think that we need to check this for every iteration